src/games/strategy/triplea/ai/proAI/util/ProMatches.java

```diff
@@ -332,6 +332,21 @@ public boolean match(final Territory t) {
     };
   }
 
+  public static Match<Territory> territoryHasNonMobileInfraFactoryAndIsLand() {
+    return new Match<Territory>() {
+      @Override
+      public boolean match(final Territory t) {
+        final Match<Unit> nonMobileinfraFactoryMatch =
+            new CompositeMatchAnd<Unit>(Matches.UnitCanProduceUnits, Matches.UnitIsInfrastructure,
+                Matches.unitHasMovementLeft.invert());
```

Comment:
.invert() makes this harder to understand, strongly recommend we go ahead and add a `Match.unitHasNoMovementLeft`